transfers/transfer.py

```diff
@@ -336,13 +342,19 @@ def start_transfer(ss_url, ss_user, ss_api_key, ts_location_uuid, ts_path, depth
         session.add(new_transfer)
         return None
 
-    # Run all scripts in pre-transfer directory
-    # TODO what inputs do we want?
-    run_scripts(
-        'pre-transfer',
-        resp_json['path'],  # Absolute path
-        'standard',  # Transfer type
-    )
+    try:
+        # Run all scripts in pre-transfer directory
+        # TODO what inputs do we want?
+        run_scripts(
+            'pre-transfer',
+            resp_json['path'],  # Absolute path
+            'standard',  # Transfer type
+        )
+    except:
+        # An error occurred, log and recover cleanly, returning to main process
+        e = sys.exc_info()[0]
+        LOGGER.error("Failed to run pre-transfer scripts: %s", e)
+        return None
```

Comment:
I think this block could be written as follows:

```python
try:
    ...
except Exception as err:
    LOGGER.error("Failed to run pre-transfer scripts: %s", err)
    return None
```

Reply:
You're right, it could, but then it wouldn't [catch general exceptions](https://wiki.python.org/moin/HandlingExceptions), only those that extend from `Exception`, so I think this way is better (my choice was deliberate).

Reply:
So why not to use `BaseException` instead? Bare excepts are generally not recommended. Bare excepts where used it in Py2 because you could have a custom exception not subclassing `Exception` but that's not possible anymore neither you have such kind of code.

Reply:
https://docs.python.org/2/howto/doanddont.html#except

Reply:
I've filed https://github.com/JiscRDSS/rdss-archivematica-automation-tools/issues/13 btw.

Reply:
Feel free to ignore this. I think I am used to avoid bare excepts because my linter complains about it but it should not be a problem.

Reply:
I've changed the `except` line to name `BareException`. I've also fixed the length of the summary line of the commit. If you're happy with these changes then please approve so we can merge.

I'm assuming by the fact you're nitpicking that you're okay with the main purpose of this commit, namely the use of the `script_extensions` whitelist and its config.